grails-app/services/org/pih/warehouse/core/DocumentTemplateService.groovy

```diff
@@ -0,0 +1,221 @@
+        else {
+            report.process(context, outputStream)
+        }
+        log.info "Report dump: " + report.dump()
+
+    }
+
+    def createOrderContext(IXDocReport report, Order orderInstance) {
+
+        // Add data to the context
+        def orderItems = orderInstance.orderItems.collect { OrderItem orderItem ->
+            return [
+                    code                : orderItem?.product?.productCode,
+                    type                : "Item",
+                    status              : orderItem.orderItemStatusCode?:"",
+                    description         : orderItem?.description ?: orderItem?.product?.name,
+                    quantity            : orderItem?.quantity ?: 0,
+                    supplierCode        : orderItem?.productSupplier?.supplierCode ?: "",
+                    manufacturer        : orderItem?.productSupplier?.manufacturer?.name ?: "",
+                    manufacturerCode    : orderItem?.productSupplier?.manufacturerCode ?: "",
+                    unitOfMeasure       : orderItem?.unitOfMeasure ?: "",
+                    unitPrice           : orderItem?.unitPrice ?: "",
+                    totalPrice          : orderItem?.totalPrice() ?: "",
+                    expectedShippingDate: orderItem?.estimatedShipDate ?: ""
+            ]
+        }
+        def orderAdjustments = orderInstance.orderAdjustments.collect { OrderAdjustment orderAdjustment ->
+            return [
+                    code                : orderAdjustment?.orderAdjustmentType?.code?:"",
+                    type                : "Adjustment",
+                    status              : "",
+                    description         : orderAdjustment?.description?:orderAdjustment?.orderAdjustmentType?.description?:"",
+                    quantity            : 1,
+                    supplierCode        : "",
+                    manufacturer        : "",
+                    manufacturerCode    : "",
+                    unitOfMeasure       : "",
+                    unitPrice           : "",
+                    totalPrice          : orderAdjustment?.totalAdjustments ?: "",
+                    expectedShippingDate: ""
+            ]
+        }
+
+        def order = orderInstance.toJson()
+        order.dateOrdered = orderInstance.dateOrdered
+        order.dateApproved = orderInstance.dateApproved
+        order.dateCompleted = orderInstance.dateCompleted
+        order.currencyCode = orderInstance.currencyCode?:""
+        order.exchangeRate = orderInstance.exchangeRate?:""
+        order.total = orderInstance.total?:""
+        order.subtotal = orderInstance.subtotal?:""
+        order.totalAdjustments = orderInstance.totalAdjustments?:""
+        order.description = orderInstance.description?:""
+
+        Address nullAddress = [address:"", address2: "", city: "", stateOrProvince: "", description: ""]
+        Organization vendor = orderInstance.originParty
+        order.vendor = vendor?.properties?:[:]
```

Comment:
I'm building a map to use as the model for the document template rendering engine.

Reply:
I'm trying to make it easier on the document desiger.

This 
```
order.vendor.address 
```
instead of
```
order.originParty.defaultLocation.address.address
```